api/pkg/s3/bucketload.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3
+
+import (
+	"errors"
+	"net/http"
+
+	"encoding/xml"
+
+	"github.com/emicklei/go-restful"
+	"github.com/micro/go-log"
+	. "github.com/opensds/multi-cloud/s3/pkg/exception"
+	"github.com/opensds/multi-cloud/s3/pkg/model"
+	"golang.org/x/net/context"
+	pb "github.com/opensds/multi-cloud/s3/proto"
+)
+
+func (s *APIService) BucketLoad(request *restful.Request, response *restful.Response) {
+	bucketName := request.PathParameter("bucketName")
+	ctx := context.Background()
+
+	body := ReadBody(request)
+	if body == nil {
+		log.Logf("received load objects of bucket[%s], but body is nil\n", bucketName)
+		response.WriteError(http.StatusBadRequest, errors.New("request body is nil"))
+		return
+	}
+
+	loadObjsReq := model.LoadObjectsReq{}
+	err := xml.Unmarshal(body, &loadObjsReq)
+	if err != nil {
+		log.Logf("received load objects of bucket[%s], but parse body failed, err:%v\n", bucketName, err)
+		response.WriteError(http.StatusBadRequest, errors.New("request body is nil"))
+	}
+
+	log.Logf("received load objects of bucket[%s], backend=%s, prefix=%s\n", bucketName, loadObjsReq.Backend, loadObjsReq.Prefix)
+
+	client := getBackendByName(s, loadObjsReq.Backend)
+	if client == nil {
+		log.Logf("get backend[name=%s] failed\n", loadObjsReq.Backend)
+		response.WriteError(http.StatusInternalServerError, NoSuchBackend.Error())
+		return
+	}
+
+	marker := ""
+	var limit, total, succeed int64 = 1000, 0, 0
```

Comment:
how did we settle on the limit of 1000 ?

Reply:
Many clouds use 1000 as the default max-keys when list objects, like AWS S3.

Reply:
AWS and GCP has default limit of 1000 but Azure has limit of 5000. But i think 1000 should be fine to settle down